### PR TITLE
feat: add support for helm skipTests flag

### PR DESCRIPTION
**Pull request by jaehanbyun**
Fixes [ISSUE #19956]

Checklist:

* [ ] Either (a) I've created an [enhancement proposal](https://github.com/argoproj/argo-cd/issues/new/choose) and discussed it with the community, (b) this is a bug fix, or (c) this does not need to be in the release notes.
* [X] The title of the PR states what changed and the related issues number (used for the release note).
* [X] The title of the PR conforms to the [Toolchain Guide](https://argo-cd.readthedocs.io/en/latest/developer-guide/toolchain-guide/#title-of-the-pr)
* [X] I've included "Closes [ISSUE #]" or "Fixes [ISSUE #]" in the description to automatically close the associated issue.
* [ ] I've updated both the CLI and UI to expose my feature, or I plan to submit a second PR with them.
* [x] Does this PR require documentation updates?
* [X] I've updated documentation as required by this PR.
* [X] I have signed off all my commits as required by [DCO](https://github.com/argoproj/argoproj/blob/master/community/CONTRIBUTING.md#legal)
* [X] I have written unit and/or e2e tests for my change. PRs without these are unlikely to be merged.
* [ ] My build is green ([troubleshooting builds](https://argo-cd.readthedocs.io/en/latest/developer-guide/ci/)).
* [X] My new feature complies with the [feature status](https://github.com/argoproj/argoproj/blob/master/community/feature-status.md) guidelines.
* [ ] I have added a brief description of why this PR is necessary and/or what this PR solves.

## What I have done:
* Added SkipTests to ApplicationSourceHelm.
* Added SkipTests to TemplateOpts.
* Passed SkipTests from ApplicationSourceHelm to TemplateOpts in helmTemplate.
* If TemplateOpts.SkipTests is set, append --skip-tests to the options generated in template.
* The --helm-skip-tests flag should now work as expected in the CLI.
* Unit tests: Write unit tests for this new feature.
* e2e tests: Write e2e tests for this new feature

### CODE REVIEW

File: docs/user-guide/helm.md [hunk 1]
```diff
@@ -498,3 +498,24 @@ spec:
     helm:
       skipCrds: true
 ```
+
+
+## Helm `--skip-tests`
+
+By default, Helm includes test manifests when rendering templates.
+Argo CD will assume that skipTests is true unless it is explicitly set to false within the Argo CD Application.
```
Comment (crenshaw-dev) [inline docs/user-guide/helm.md:506]: If I understand correctly, `--skip-tests` is `false` by default in Helm. So changing the default in Argo CD to `true` would be a breaking change. If I'm correct, then I think we should switch the default to `false`, even if it's a somewhat less-nice UX.

File: docs/user-guide/helm.md [hunk 2]
```diff
@@ -498,3 +498,24 @@ spec:
...
+argocd app set helm-guestbook --helm-skip-tests false
+```
+
+Or using declarative syntax:
+
+```yaml
+spec:
+  source:
+    helm:
+       skipTests: true # or false
```
Comment (crenshaw-dev) [inline docs/user-guide/helm.md:520]:
```suggestion
      skipTests: true # or false
```

**Review by crenshaw-dev — CHANGES_REQUESTED** [review]
Nicely coded! Added one comment about the default value.

File: docs/user-guide/helm.md (hunk 1 above)
Comment (jaehanbyun) [reply docs/user-guide/helm.md]: I've made the changes to adjust the default setting of --skip-tests to false in Argo CD. Please take a look when you have time and let me know if any further adjustments are needed.

**Review by crenshaw-dev — APPROVED** [review]

**Comment by xrow** [comment]
`one or more objects failed to apply, reason: failed to create typed patch object (argo-cd/apps; argoproj.io/v1alpha1, Kind=ApplicationSet): .spec.template.spec.sources[0].helm.skipTests: field not declared in schema,failed to create typed patch object (argo-cd/cluster; argoproj.io/v1alpha1, Kind=ApplicationSet): .spec.template.spec.sources[0].helm.skipTests: field not declared in schema`


@jaehanbyun Is it as planned that this flag can`t be used in `sources[X].helm.skipTests` ?

**Comment by xrow** [comment]
@jaehanbyun Uhh I am using the argo operator version 0.13.0 Is there a change needed to?

**Comment by jaehanbyun** [comment]
@xrow it appears that the skipTests flag is supported starting with Argo CD v2.14.0. 
Could you please confirm which version of Argo CD is being used with Argo Operator v0.13.0?

**Comment by xrow** [comment]
@jaehanbyun Ok makes sense now. The operator is behind a lot: v2.13.1+af54ef8. Thanks for telling me the version.

**Pull request merged**